### ISSUE TITLE
False negative `UseArrayLiteralsInAnnotations`

### ISSUE THREAD

**Issue by t-kameyama** [detekt/detekt#5481]
## Steps to Reproduce
```kotlin
@Test
fun `finds intArrayOf usage`() {
    val code = """
        annotation class Test(val values: IntArray)
        
        @Test(intArrayOf(1, 2))
        fun test() {}
    """.trimIndent()
    subject.compileAndLint(code)
    Assertions.assertThat(subject.findings).hasSize(1)
}

@Test
fun `finds longArrayOf usage`() {
    val code = """
        annotation class Test(val values: LongArray)
        
        @Test(longArrayOf(1, 2))
        fun test() {}
    """.trimIndent()
    subject.compileAndLint(code)
    Assertions.assertThat(subject.findings).hasSize(1)
}

@Test
fun `finds arrayOf usage as default value`() {
    val code = """
        annotation class Test(val s: Array<String> = arrayOf("a", "b"))
    """.trimIndent()
    subject.compileAndLint(code)
    Assertions.assertThat(subject.findings).hasSize(1)
}
```
